tests/data/sentinel2/data.py

```diff
@@ -12,24 +13,89 @@
+                    "T16TFM_20220412T162841_TCI.jp2",
+                ]
+            }
+        }
+    },
+    # Copernicus Open Access Hub
+    "S2A_MSIL2A_20220414T110751_N0400_R108_T26EMU_20220414T165533.SAFE": {
+        "GRANULE": {
+            "L2A_T26EMU_A035569_20220414T110747": {
+                "IMG_DATA": {
+                    "R10m": [
+                        "T26EMU_20220414T110751_AOT_10m.jp2",
+                        "T26EMU_20220414T110751_B02_10m.jp2",
+                        "T26EMU_20220414T110751_B03_10m.jp2",
+                        "T26EMU_20220414T110751_B04_10m.jp2",
+                        "T26EMU_20220414T110751_B08_10m.jp2",
+                        "T26EMU_20220414T110751_TCI_10m.jp2",
+                        "T26EMU_20220414T110751_WVP_10m.jp2",
+                    ],
+                    "R20m": [
+                        "T26EMU_20220414T110751_AOT_20m.jp2",
+                        "T26EMU_20220414T110751_B01_20m.jp2",
+                        "T26EMU_20220414T110751_B02_20m.jp2",
+                        "T26EMU_20220414T110751_B03_20m.jp2",
+                        "T26EMU_20220414T110751_B04_20m.jp2",
+                        "T26EMU_20220414T110751_B05_20m.jp2",
+                        "T26EMU_20220414T110751_B06_20m.jp2",
+                        "T26EMU_20220414T110751_B07_20m.jp2",
+                        "T26EMU_20220414T110751_B11_20m.jp2",
+                        "T26EMU_20220414T110751_B12_20m.jp2",
+                        "T26EMU_20220414T110751_B8A_20m.jp2",
+                        "T26EMU_20220414T110751_SCL_20m.jp2",
+                        "T26EMU_20220414T110751_TCI_20m.jp2",
+                        "T26EMU_20220414T110751_WVP_20m.jp2",
+                    ],
+                    "R60m": [
+                        "T26EMU_20220414T110751_AOT_60m.jp2",
+                        "T26EMU_20220414T110751_B01_60m.jp2",
+                        "T26EMU_20220414T110751_B02_60m.jp2",
+                        "T26EMU_20220414T110751_B03_60m.jp2",
+                        "T26EMU_20220414T110751_B04_60m.jp2",
+                        "T26EMU_20220414T110751_B05_60m.jp2",
+                        "T26EMU_20220414T110751_B06_60m.jp2",
+                        "T26EMU_20220414T110751_B07_60m.jp2",
+                        "T26EMU_20220414T110751_B09_60m.jp2",
+                        "T26EMU_20220414T110751_B11_60m.jp2",
+                        "T26EMU_20220414T110751_B12_60m.jp2",
+                        "T26EMU_20220414T110751_B8A_60m.jp2",
+                        "T26EMU_20220414T110751_SCL_60m.jp2",
+                        "T26EMU_20220414T110751_TCI_60m.jp2",
+                        "T26EMU_20220414T110751_WVP_60m.jp2",
+                    ],
+                }
+            }
+        }
+    },
+}
 
 
 def create_file(path: str, dtype: str, num_channels: int) -> None:
     profile = {}
-    profile["driver"] = "GTiff"
+    profile["driver"] = "JP2OpenJPEG"
```

Comment:
Why doesn't this work? With `GTiff`, rasterio happily writes GeoTIFF files with the `.jp2` extension. When I change to `JP2OpenJPEG`, rasterio creates all of the directories, but doesn't write any files to disk. There is no error message. I've reproduced this with both Spack- and Conda-installed GDAL/rasterio. GDAL itself seems fine. `gdalinfo --formats` lists this driver under both read and write. `gdal_create -of JP2OpenJPEG` works fine. Quite confused...